File: cc150/Solution1Q2.c
# include <string.h>
void reverse(char *s) {// or char s[]
    int length = strlen(s);
    int c, i, j;
    
    for (i = 0, j = length - 1; i < j; i++, j--) {
        c = s[i];
        s[i] = s[j];
        s[j] = c;
    }
}

File: cc150/Solution1Q2.java
public class Solution1Q2 {
    public String reverse(String str) {
        char[] chars = str.toCharArray();
        int start = 0;
        int end = chars.length - 1;

        while (start < end) {
            char temp = chars[start];
            chars[start] = chars[end];
            chars[end] = temp;	
            start++;
            end--;
        }

        return new String(chars);
    }
}
